src/main.rs

```diff
@@ -15,74 +14,23 @@ fn main() -> Result<(), Box<dyn Error>> {
         process::exit(0);
     }
 
-    let outputs = dotenv_linter::run(&args, &current_dir)?;
-
-    if outputs.is_empty() {
-        process::exit(0);
-    }
-
     let no_color = args.is_present("no-color");
     if no_color {
         colored::control::set_override(false);
     }
     #[cfg(windows)]
     set_windows_virtual_terminal();
-    let total = outputs.iter().map(|o| o.warnings.len()).sum();
-    let is_not_quiet = !args.is_present("quiet");
-    if args.is_present("fix") {
-        if is_not_quiet {
-            dotenv_linter::print_outputs(outputs);
-        } else {
-            outputs.iter().for_each(|w| w.print_backup());
-        }
-        print_fix_total(total);
-        process::exit(0);
-    }
 
-    if is_not_quiet {
-        dotenv_linter::print_outputs(outputs);
-        print_check_total(total);
-    } else {
-        outputs.iter().for_each(|w| w.print_warnings());
-    }
+    let warnings = dotenv_linter::run(&args, &current_dir)?;
```

Comment:
Maybe it's better to return the number of warnings instead of an array of warnings?